packages/in_app_purchase/ios/Classes/FIAPaymentQueueHandler.m

```diff
@@ -81,7 +81,11 @@ - (void)paymentQueue:(SKPaymentQueue *)queue
       //    will become impossible for clients to finish deferred transactions when needed.
       // 2. Using product identifiers can help prevent clients from purchasing the same
       //    subscription more than once by accident.
-      self.transactionsSetter[transaction.payment.productIdentifier] = transaction;
+      if ([self.transactionsSetter objectForKey:transaction.payment.productIdentifier] == nil) {
```

Comment:
This `== nil` is unnecessary.

Reply:
How so? You only want to init a new array to that key if there aren't an array with transactions for that key already.